File: studier_flutter/lib/components/organisms/tutor_header.dart
import 'package:flutter/material.dart';
import '../../models/tutor_model.dart';
import '../../utils/constants.dart';
import '../atoms/avatar.dart';
import '../atoms/star_rating.dart';

/// Organism: The header section of a tutor profile page.
class TutorHeader extends StatelessWidget {
  final Tutor tutor;
  final double? averageRatingOverride;
  final int? totalReviewsOverride;
  final bool isDark;

  const TutorHeader({
    super.key,
    required this.tutor,
    this.averageRatingOverride,
    this.totalReviewsOverride,
    this.isDark = false,
  });

  @override
  Widget build(BuildContext context) {
    final averageRating = averageRatingOverride ?? tutor.averageRating;
    final totalReviews = totalReviewsOverride ?? tutor.totalReviews;
    final backgroundColor = isDark ? AppColors.backgroundDark : AppColors.white;
    final primaryTextColor = isDark ? AppColors.white : AppColors.gray900;
    final secondaryTextColor = isDark ? AppColors.gray300 : AppColors.gray700;
    final tertiaryTextColor = isDark ? AppColors.gray400 : AppColors.gray500;
    final chipColor = isDark
        ? AppColors.surfaceDark
        : AppColors.backgroundLight;
    final chipTextColor = isDark ? AppColors.gray300 : AppColors.gray700;
    final dividerColor = isDark
        ? Colors.white.withValues(alpha: 0.08)
        : AppColors.gray100;

    return Container(
      padding: const EdgeInsets.all(16),
      color: backgroundColor,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          // Avatar row
          Row(
            children: [
              Avatar(imageUrl: tutor.avatarUrl, radius: 50),
              const SizedBox(width: 16),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Row(
                      children: [
                        Flexible(
                          child: Text(
                            tutor.name,
                            style: TextStyle(
                              fontSize: 20,
                              fontWeight: FontWeight.w700,
                              color: primaryTextColor,
                            ),
                          ),
                        ),
                        if (tutor.verified) ...[
                          const SizedBox(width: 6),
                          const Icon(
                            Icons.verified,
                            size: 18,
                            color: AppColors.primary,
                          ),
                        ],
                      ],
                    ),
                    const SizedBox(height: 2),
                    Text(
                      tutor.university,
                      style: TextStyle(fontSize: 14, color: secondaryTextColor),
                    ),
                    if (tutor.major.trim().isNotEmpty)
                      Text(
                        tutor.major,
                        style: TextStyle(
                          fontSize: 13,
                          color: tertiaryTextColor,
                        ),
                      ),
                  ],
                ),
              ),
            ],
          ),

          const SizedBox(height: 12),

          // Bio
          _detailLine('Profession/Roles', tutor.profession),
          const SizedBox(height: 6),
          _detailLine('Education', tutor.education),
          const SizedBox(height: 6),
          _detailLine('Expertise/Interests', tutor.subjectTags.join(', ')),
          const SizedBox(height: 6),
          _detailLine('Bio', tutor.bio),

          const SizedBox(height: 12),

          // Subject tags
          Wrap(
            spacing: 8,
            runSpacing: 8,
            children: tutor.subjectTags.map((tag) {
              return Container(
                padding: const EdgeInsets.symmetric(
                  horizontal: 10,
                  vertical: 6,
                ),
                decoration: BoxDecoration(
                  color: chipColor,
                  borderRadius: BorderRadius.circular(12),
                  border: Border.all(
                    color: isDark
                        ? Colors.white.withValues(alpha: 0.06)
                        : Colors.transparent,
                  ),
                ),
                child: Text(
                  tag,
                  style: TextStyle(
                    fontSize: 12,
                    fontWeight: FontWeight.w600,
                    color: chipTextColor,
                  ),
                ),
              );
            }).toList(),
          ),

          const SizedBox(height: 16),

          // Stats row
          Container(
            padding: const EdgeInsets.symmetric(vertical: 12),
            decoration: BoxDecoration(
              border: Border(top: BorderSide(color: dividerColor)),
            ),
            child: Row(
              children: [
                _StatTile(
                  value: '${tutor.totalSessions}',
                  label: 'Sessions',
                  isDark: isDark,
                ),
                _divider(),
                _StatTile(
                  value: '${averageRating.toStringAsFixed(1)}',
                  label: 'Rating',
                  trailing: StarRating(rating: averageRating, size: 12),
                  isDark: isDark,
                ),
                _divider(),
                _StatTile(
                  value: totalReviews.toString(),
                  label: 'Reviews',
                  isDark: isDark,
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _detailLine(String label, String value) {
    final text = value.trim().isEmpty ? '-' : value.trim();
    return Text(
      '$label: $text',
      style: TextStyle(
        fontSize: 14,
        color: isDark ? AppColors.gray300 : AppColors.gray700,
        height: 1.4,
      ),
    );
  }

  Widget _divider() => Container(
    width: 1,
    height: 32,
    color: isDark ? Colors.white.withValues(alpha: 0.08) : AppColors.gray100,
  );
}

class _StatTile extends StatelessWidget {
  final String value;
  final String label;
  final Widget? trailing;
  final bool isDark;

  const _StatTile({
    required this.value,
    required this.label,
    this.trailing,
    this.isDark = false,
  });

  @override
  Widget build(BuildContext context) {
    return Expanded(
      child: Column(
        children: [
          if (trailing != null)
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                trailing!,
                const SizedBox(width: 4),
                Text(
                  value,
                  style: TextStyle(
                    fontSize: 16,
                    fontWeight: FontWeight.w700,
                    color: isDark ? AppColors.white : AppColors.gray900,
                  ),
                ),
              ],
            )
          else
            Text(
              value,
              style: TextStyle(
                fontSize: 16,
                fontWeight: FontWeight.w700,
                color: isDark ? AppColors.white : AppColors.gray900,
              ),
            ),
          const SizedBox(height: 4),
          Text(
            label,
            style: TextStyle(
              fontSize: 12,
              color: isDark ? AppColors.gray400 : AppColors.gray500,
            ),
          ),
        ],
      ),
    );
  }
}

File: src/components/tutor/TutorHeader.js
import React from 'react';
import { View, Text, StyleSheet, Image } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import { COLORS } from '../../utils/constants';
import StarRating from '../common/StarRating';

const TutorHeader = ({ tutor }) => {
  if (!tutor) {
    return null;
  }

  return (
    <View style={styles.container}>
      <View style={styles.avatarRow}>
        <Image source={{ uri: tutor.image }} style={styles.avatar} />
        <View style={styles.infoColumn}>
          <View style={styles.nameRow}>
            <Text style={styles.name}>{tutor.name}</Text>
            {tutor.verified && (
              <MaterialIcons name="verified" size={16} color={COLORS.primary} />
            )}
          </View>
          <Text style={styles.university}>{tutor.university}</Text>
          <Text style={styles.year}>{tutor.year}</Text>
        </View>
      </View>

      <Text style={styles.bio} numberOfLines={3}>
        {tutor.bio}
      </Text>

      <View style={styles.tagsRow}>
        {tutor.subjects.map((subject) => (
          <View key={subject} style={styles.tag}>
            <Text style={styles.tagText}>{subject}</Text>
          </View>
        ))}
      </View>

      <View style={styles.statsRow}>
        <View style={styles.statItem}>
          <Text style={styles.statValue}>{tutor.sessions}</Text>
          <Text style={styles.statLabel}>Sessions</Text>
        </View>
        <View style={styles.statDivider} />
        <View style={styles.statItem}>
          <View style={styles.ratingRow}>
            <StarRating rating={tutor.rating} size={14} />
            <Text style={styles.statValue}>{tutor.rating}</Text>
          </View>
          <Text style={styles.statLabel}>Rating</Text>
        </View>
        <View style={styles.statDivider} />
        <View style={styles.statItem}>
          <Text style={styles.statValue}>{tutor.reviews}</Text>
          <Text style={styles.statLabel}>Reviews</Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
    paddingTop: 16,
    paddingBottom: 8,
    backgroundColor: COLORS.white,
  },
  avatarRow: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 16,
    marginBottom: 12,
  },
  avatar: {
    width: 100,
    height: 100,
    borderRadius: 50,
    backgroundColor: COLORS.gray[200],
  },
  infoColumn: {
    flex: 1,
  },
  nameRow: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 6,
    marginBottom: 4,
  },
  name: {
    fontSize: 20,
    fontWeight: '700',
    color: COLORS.gray[900],
  },
  university: {
    fontSize: 14,
    color: COLORS.gray[700],
  },
  year: {
    fontSize: 13,
    color: COLORS.gray[500],
    marginTop: 4,
  },
  bio: {
    fontSize: 14,
    color: COLORS.gray[700],
    lineHeight: 20,
    marginBottom: 12,
  },
  tagsRow: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    gap: 8,
    marginBottom: 16,
  },
  tag: {
    backgroundColor: COLORS.backgroundLight,
    borderRadius: 12,
    paddingHorizontal: 10,
    paddingVertical: 6,
  },
  tagText: {
    fontSize: 12,
    color: COLORS.gray[700],
    fontWeight: '600',
  },
  statsRow: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: 12,
    borderTopWidth: 1,
    borderTopColor: COLORS.gray[100],
  },
  statItem: {
    flex: 1,
    alignItems: 'center',
    gap: 4,
  },
  statDivider: {
    width: 1,
    height: 32,
    backgroundColor: COLORS.gray[100],
  },
  statValue: {
    fontSize: 16,
    fontWeight: '700',
    color: COLORS.gray[900],
  },
  statLabel: {
    fontSize: 12,
    color: COLORS.gray[500],
  },
  ratingRow: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 6,
  },
});

export default TutorHeader;
